From 783cd805a7413a7d415ab97a2857b0d925d5cc08 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sun, 4 Nov 2018 13:09:16 +0100
Subject: [PATCH] Do not assume timestamps are strictly monotonous
Forwarded: https://github.com/codership/galera/pull/526

Some OS do not provide sub-tick time granularity, and ticks can typically be
10ms.
---
 gcs/src/unit_tests/gcs_sm_test.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcs/src/unit_tests/gcs_sm_test.cpp b/gcs/src/unit_tests/gcs_sm_test.cpp
index 839e21c4..14a3dbd2 100644
--- a/gcs/src/unit_tests/gcs_sm_test.cpp
+++ b/gcs/src/unit_tests/gcs_sm_test.cpp
@@ -280,7 +280,7 @@ START_TEST (gcs_sm_test_pause)
     long long tmp;
     gcs_sm_stats_get (sm, &q_len, &q_len_max, &q_len_min, &q_len_avg,
                       &tmp, &paused_avg);
-    fail_if (tmp <= paused_ns); paused_ns = tmp;
+    fail_if (tmp < paused_ns); paused_ns = tmp;
     fail_if (paused_avg <= 0.0);
     fail_if (fabs(q_len_avg) > EPS,
              "q_len_avg: expected <= %e, got %e", EPS, fabs(q_len_avg));
